doc/source/development/developer.rst

```diff
@@ -53,26 +60,43 @@ for each column, *including the index columns*. This has JSON form:
     'numpy_type': numpy_type,
     'metadata': metadata}
 
-.. note::
+See below for the detailed specification for these
+
+Index Metadata Descriptors
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+``RangeIndex`` can be stored as metadata only, not requiring serialization. The
+descriptor format for these as is follows:
+
+.. code-block:: python
+
+   index = pd.RangeIndex(0, 10, 2)
+   {'kind': 'range',
+    'name': index.name,
+    'start': index._start,
```

Comment:
https://github.com/pandas-dev/pandas/pull/25720 was just merged, prob could just change this (though of course implementing this for <0.25.x will still require this)